pkg/jobs/workers/thumbnail.go

```diff
@@ -27,30 +38,93 @@ func init() {
 
 // ThumbnailWorker is a worker that creates thumbnails for photos and images.
 func ThumbnailWorker(ctx context.Context, m *jobs.Message) error {
-	domain := ctx.Value(jobs.ContextDomainKey).(string)
 	msg := &imageMessage{}
 	if err := m.Unmarshal(msg); err != nil {
 		return err
 	}
-	fmt.Printf("[jobs] thumbnails %s: %#v", domain, msg)
+	domain := ctx.Value(jobs.ContextDomainKey).(string)
+	i, err := instance.Get(domain)
+	if err != nil {
+		return err
+	}
 	switch msg.Event.Type {
 	case "CREATED":
-		return generateThumbnails(domain, msg.Event.Doc)
+		return generateThumbnails(ctx, i, msg.Event.Doc)
 	case "UPDATED":
-		if err := removeThumbnails(domain, msg.Event.Doc); err != nil {
+		if err = removeThumbnails(i, msg.Event.Doc); err != nil {
 			return err
 		}
-		return generateThumbnails(domain, msg.Event.Doc)
+		return generateThumbnails(ctx, i, msg.Event.Doc)
 	case "DELETED":
-		return removeThumbnails(domain, msg.Event.Doc)
+		return removeThumbnails(i, msg.Event.Doc)
 	}
 	return fmt.Errorf("Unknown type %s for image event", msg.Event.Type)
 }
 
-func generateThumbnails(domain string, image vfs.FileDoc) error {
-	return nil
+func generateThumbnails(ctx context.Context, i *instance.Instance, img vfs.FileDoc) error {
+	fs := i.ThumbsFS()
+	if err := fs.MkdirAll(thumbDir(img), 0755); err != nil {
+		return err
+	}
+	flags := os.O_RDWR | os.O_CREATE
+	in, err := i.VFS().OpenFile(&img)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+	largeName := thumbName(img, "large")
+	large, err := fs.OpenFile(largeName, flags, 0640)
+	if err != nil {
+		return err
+	}
+	defer large.Close()
```

Comment:
since this serie of `generateThumb` calls can take quite some time, I would have the `generateThumb` method do the call to the `Close()` method in order to have less file handlers laying around.